Python/intrinsics.c

```diff
@@ -0,0 +1,186 @@
+            _PyErr_SetString(tstate, PyExc_ImportError,
+                    "from-import-* object has no __dict__ and no __all__");
+            return -1;
+        }
+        all = PyMapping_Keys(dict);
+        Py_DECREF(dict);
+        if (all == NULL)
+            return -1;
+        skip_leading_underscores = 1;
+    }
+
+    for (pos = 0, err = 0; ; pos++) {
+        name = PySequence_GetItem(all, pos);
+        if (name == NULL) {
+            if (!_PyErr_ExceptionMatches(tstate, PyExc_IndexError)) {
+                err = -1;
+            }
+            else {
+                _PyErr_Clear(tstate);
+            }
+            break;
+        }
+        if (!PyUnicode_Check(name)) {
+            PyObject *modname = PyObject_GetAttr(v, &_Py_ID(__name__));
+            if (modname == NULL) {
+                Py_DECREF(name);
+                err = -1;
+                break;
+            }
+            if (!PyUnicode_Check(modname)) {
+                _PyErr_Format(tstate, PyExc_TypeError,
+                              "module __name__ must be a string, not %.100s",
+                              Py_TYPE(modname)->tp_name);
+            }
+            else {
+                _PyErr_Format(tstate, PyExc_TypeError,
+                              "%s in %U.%s must be str, not %.100s",
+                              skip_leading_underscores ? "Key" : "Item",
+                              modname,
+                              skip_leading_underscores ? "__dict__" : "__all__",
+                              Py_TYPE(name)->tp_name);
+            }
+            Py_DECREF(modname);
+            Py_DECREF(name);
+            err = -1;
+            break;
+        }
+        if (skip_leading_underscores) {
+            if (PyUnicode_READY(name) == -1) {
+                Py_DECREF(name);
+                err = -1;
+                break;
+            }
+            if (PyUnicode_READ_CHAR(name, 0) == '_') {
+                Py_DECREF(name);
+                continue;
+            }
+        }
+        value = PyObject_GetAttr(v, name);
+        if (value == NULL)
+            err = -1;
+        else if (PyDict_CheckExact(locals))
+            err = PyDict_SetItem(locals, name, value);
+        else
+            err = PyObject_SetItem(locals, name, value);
+        Py_DECREF(name);
+        Py_XDECREF(value);
+        if (err != 0)
+            break;
+    }
+    Py_DECREF(all);
+    return err;
+}
+
+static PyObject *
+import_star(PyThreadState* tstate, PyObject *from)
+{
+    _PyInterpreterFrame *frame = tstate->cframe->current_frame;
+    if (_PyFrame_FastToLocalsWithError(frame) < 0) {
+        return NULL;
+    }
+
+    PyObject *locals = frame->f_locals;
+    if (locals == NULL) {
+        _PyErr_SetString(tstate, PyExc_SystemError,
+                            "no locals found during 'import *'");
+        return NULL;
+    }
+    int err = import_all_from(tstate, locals, from);
+    _PyFrame_LocalsToFast(frame, 0);
+    if (err) {
```

Comment:
In import_star you have "if (err != 0)", here just "if (err)". In other places we check "if (err < 0)".